Cython/Compiler/ExprNodes.py

```diff
@@ -3140,6 +3148,8 @@ def type_dependencies(self, env):
 
     def infer_type(self, env):
         sequence_type = self.sequence.infer_type(env)
+        if sequence_type in subscription_aware_types and (sub_type := sequence_type.get_subscribed_type(0)):
+            return sub_type
```

Comment:
The name `sub_type` suggests "subclass type". Let's use "item type" as the term for this.
```suggestion
        if sequence_type in subscription_aware_types:
            item_type = sequence_type.get_subscribed_type(0)
            if item_type:
                return item_type
```

Reply:
fixed.